doc/HttpTransportAdapter.md

```diff
@@ -0,0 +1,192 @@
+# HTTP Transport Adapter
+
+The Azure SDK for C++ creates and sends HTTP requests using an HTTP Transport Adapter. This adapter provides a general abstraction for an HTTP stack (like libcurl, WinHTTP, etc) by defining the interface it needs to call that anyone can implement to customize which HTTP stack gets used.
+
+## Default HTTP Transport Adapter
+The Azure SDK for C++ provides two HTTP transport adapters as part of the azure-core package:
+- The libcurl transport adapter
+- WinHttp transport adapter
+
+Azure SDK for C++ users can create and use their own transport adapter as well [by following the guidance below](#building-a-custom-http-transport-adapter).
+
+For the simplest case, when no specific configuration is set, the default transport adapter will be selected based on the OS. The WinHTTP transport adapter will be used for Windows-based systems (making libcurl an optional dependency). The libcurl transport adapter will be selected for any non-windows-based system.
+
+## Building HTTP transport adapter
+
+The Azure SDK for C++ uses CMake options to define what HTTP transport adapters to build.
+
+You can see [CMake options](https://github.com/Azure/azure-sdk-for-cpp/blob/master/CONTRIBUTING.md#cmake-build-options) to learn about all the supported options and what is the specific option required for each HTTP transport adapter.
+
+Multiple HTTP transport adapters can be built as part of the project. This is to support scenarios where you want to send HTTP request with the libcurl transport adapter for some cases and the WinHTTP transport adapter for others.
+Note WinHTTP transport adapter is only supported on Windows.
+
+Another example is if you want to create your own HTTP transport adapter for some other C++ HTTP stack. In this case, you can build your own HTTP transport adapter and test it by comparing it to the behavior of the libcurl HTTP transport adapter.
+
+## Using the HTTP transport adapter
+The HTTP transport adapter is set up during service SDK client initialization, for example, when creating an Azure Storage SDK client. The HTTP transport adapter can be specified at client initialization via the client options argument. See the following code snippet as an example:
+```cpp
+  /*
+  * Using the HTTP transport adapter.
+  */
+
+  // When no options are provided, the HTTP transport adapter is set to the default one based on the compiler options passed in to CMake.
+  auto defaultStorageClient = BlobServiceClient(url, credential);
+
+  // Use client options to set an HTTP transport adapter, which needs to be a shared_ptr.
```

Comment:
Break the example into two and separate with this text:
This example below shows how to override the default HTTP transport adapter when constructing a client: